convert2rhel/utils.py

```diff
@@ -158,10 +158,6 @@ def run_subprocess(cmd, print_cmd=True, print_output=True):
         if print_output:
             loggerinst.info(line.rstrip("\n"))
 
-    # Call communicate() to wait for the process to terminate so that we can get the return code by poll().
-    # It's just for py2.6, py2.7+/3 doesn't need this.
-    process.communicate()
-
```

Comment:
@bocekm this seems to still be needed for py2.7 and py3. Both unit tests fail due to this removal. Do you know why that is before I restore?

Reply:
True. The comment was wrong. When looking at the documentation of `poll()` it doesn't wait for the process to finish. `wait()` does but there's a potential deadlock when using `stdout=PIPE` which we do, so it's better to wait for the process to finish through calling `communicate()` and then `poll()` to get the return code.

Reply:
Done